antarest/storage/service.py

```diff
@@ -1,328 +1,112 @@
-        name = (
-            f"-{info_antares_output['name']}"
-            if info_antares_output["name"]
-            else ""
-        )
-
-        output_name = f"{date}{mode}{name}"
-        path_output.rename(Path(path_output.parent, output_name))
-
-        output_id = (
-            sorted(os.listdir(path_output.parent)).index(output_name) + 1
-        )
-
-        data = self.get(
-            f"{uuid}/output/{output_id}",
-            parameters=StorageServiceParameters(depth=-1),
-        )
-
-        if data is None:
-            self.delete_output(uuid, "imported_output")
-            raise BadOutputError("The output provided is not conform.")
-
-        return data
+        return self.importer_service.import_output(uuid, stream)
 
     def edit_study(self, route: str, new: JSON) -> JSON:
-        # Get data
-        uuid, url, study_path = self._extract_info_from_url(route)
-        self.assert_study_exist(uuid)
-
-        _, study = self.study_factory.create_from_fs(study_path)
-        study.save(new, url.split("/"))
-        del study
-        return new
-
-    @staticmethod
-    def check_antares_version(study: JSON) -> None:
-
-        version = study["study"]["antares"]["version"]
-        major_version = int(version / 100)
-
-        if major_version < 7:
-            raise StudyValidationError(
-                "The API do not handle study with antares version inferior to 7"
-            )
-
-    @staticmethod
-    def generate_uuid() -> str:
-        return str(uuid4())
-
-    @staticmethod
-    def extract_zip(stream: IO[bytes], dst: Path) -> None:
-        try:
-            with ZipFile(stream) as zip_output:
-                zip_output.extractall(path=dst)
-        except BadZipFile:
-            raise BadZipBinary("Only zip file are allowed.")
-
-    @staticmethod
-    def assert_path_can_be_matrix(path: Path) -> None:
-        if path.suffix != ".txt":
-            raise IncorrectPathError(
-                f"{path} is not a valid path for a matrix (use txt extension)."
-            )
-
-    @staticmethod
-    def _update_antares_info(study_name: str, study_data: JSON) -> None:
-        # TODO return value rather than change implicitly
-        info_antares = study_data["study"]["antares"]
-
-        info_antares["caption"] = study_name
-        current_time = int(time.time())
-        info_antares["created"] = current_time
-        info_antares["lastsave"] = current_time
-
-    def _extract_info_from_url(self, route: str) -> Tuple[str, str, Path]:
-        route_parts = route.split("/")
-        uuid = route_parts[0]
-        url = "/".join(route_parts[1:])
-        study_path = self.path_to_studies / uuid
-
-        return uuid, url, study_path
+        return self.study_service.edit_study(route, new)
+
+    def _check_user_permission(self, user: User, uuid: str) -> None:
+        md = self.repository.get(uuid)
+        if not md:
+            raise StudyNotFoundError(uuid)
```

Comment:
log a warning indicating that study metadata was not found and add a TODO for metadata autocreation from study files.
this will not raise any exception